content/en/blog/_posts/2022-11-29-pod-scheduling-readiness/index.md

```diff
@@ -0,0 +1,71 @@
+---
+layout: blog
+title: "Kubernetes 1.26: Pod Scheduling Readiness to Alpha"
+date: 2022-11-29
+slug: pod-scheduling-readiness-alpha-1-26
+---
+
+**Author:** Wei Huang (Apple), Abdullah Gharaibeh (Google)
+
+Kubernetes 1.26 introduced a new Pod feature named SchedulingGates. SchedulingGates are keys that
+tell the scheduler when a Pod is ready to be considered for scheduling.
+
+## What problem does it solve?
+
+When a Pod is created, the scheduler will continuously attempt to find a node that fits it. This
+infinite loop continues until the scheduler either finds a node for the Pod, or the Pod gets deleted.
+
+Pods that remain unschedulable for long periods of time (e.g., one that is blocked on some external event) 
+waste scheduling cycles. A scheduling cycle may take ~20ms or more depending on the complexity of
+the Pod's scheduling constraints. Therefore, at scale, those wasted cycles significantly impact the scheduler's performance. See the cycle in red arrows below.
+
+![](./scheduler-workflow-1.png)
+
+SchedulingGates helps address this problem. It allows declaring that newly created Pods are not
+ready for scheduling. When SchedulingGates are present on a Pod, the scheduler ignores the Pod
+and therefore saves unnecessary scheduling attempts. Clearing the gates is the responsibility of
+external controllers with knowledge of when the Pod should be considered for scheduling (e.g., 
+a resource provisioner).
+
+![](scheduler-workflow-2.png)
+
+## How does it work?
+
+SchedulingGates in general works very similar to Finalzers. Pods with non-empty 
+`spec.schedulingGates` field will carry the status `SchedulingGated` and will be blocked from
```

Comment:
Is this right:
```suggestion
`spec.schedulingGates` field will gain the `SchedulingGated`
[condition](/docs/concepts/workloads/pods/pod-lifecycle/#pod-conditions) and will be blocked from
```

Reply:
Not quite, as it's not a new condition. The condition itself is still of type `PodScheduled`, just with a new reason `SchedulingGated`. How about rewording as "... will show as status ..."?

Reply:
> type `PodScheduled`, just with a new reason `SchedulingGated`

That sounds like good wording! You can also explain how `kubectl` represents this.